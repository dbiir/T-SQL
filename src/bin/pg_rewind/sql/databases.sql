#!/bin/bash

# This file has the .sql extension, but it is actually launched as a shell
# script. This contortion is necessary because pg_regress normally uses
# psql to run the input scripts, and requires them to have the .sql
# extension, but we use a custom launcher script that runs the scripts using
# a shell instead.

TESTNAME=databases

. sql/config_test.sh

# Create a database in master.
function before_standby
{
PGOPTIONS=${PGOPTIONS_UTILITY} $MASTER_PSQL <<EOF
CREATE DATABASE inmaster;
EOF
}

function standby_following_master
{
# Create another database after promotion
PGOPTIONS=${PGOPTIONS_UTILITY} $MASTER_PSQL -c "CREATE DATABASE beforepromotion"
}

# This script runs after the standby has been promoted. Old Master is still
# running.
function after_promotion
{
PGOPTIONS=${PGOPTIONS_UTILITY} $MASTER_PSQL -c "CREATE DATABASE master_afterpromotion"

PGOPTIONS=${PGOPTIONS_UTILITY} $STANDBY_PSQL -c "CREATE DATABASE standby_afterpromotion"
}

# Compare results generated by querying original master after rewind
function after_rewind
{
PGOPTIONS=${PGOPTIONS_UTILITY} $MASTER_PSQL -c "SELECT datname from pg_database"
}
# Run the test
. sql/run_test.sh
