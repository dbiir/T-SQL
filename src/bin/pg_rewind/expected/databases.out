Master initialized.
Master running.
CREATE DATABASE inmaster;
CREATE DATABASE
Standby initialized and running.
CREATE DATABASE beforepromotion
CREATE DATABASE
Standby promoted.
CREATE DATABASE master_afterpromotion
CREATE DATABASE
CREATE DATABASE standby_afterpromotion
CREATE DATABASE
Running pg_rewind...
Old master restarted after rewind.
SELECT state FROM pg_stat_replication;
   state   
-----------
 streaming
(1 row)

Master promoted.
SELECT datname from pg_database
        datname         
------------------------
 template1
 template0
 postgres
 inmaster
 beforepromotion
 standby_afterpromotion
(6 rows)

